### DISCUSSION TITLE
Why a getter can't call a action in typescript?

### DISCUSSION THREAD

**Discussion by zh-pan** [vuejs/pinia#1390] [Help and Questions]
code like this:
```
{
getters:{
  something():{
     this.stop();// will warn
     .....
  }
},
actions:{
    stop():{
         
     }
}
}
```

If I call the `stop` func in the `something`, vscode will be display red line for warn. And run `yarn build` will be fail, but run `yarn build-only` will be success.(vite + typescript + vue3 + pinia)

The warn like this:
  Property 'stop' does not exist on type '...'

**Comment by zh-pan** [comment]
I see the answer: https://github.com/vuejs/pinia/discussions/450#discussioncomment-660238

But if this is the case, when I need get some 'like button' object, and when anyone button clicked, need call some action.

**Reply by posva** [reply]
getters are like computed properties, the same [Best practices](https://vuejs.org/guide/essentials/computed.html#writable-computed) apply so: no side effects. It sounds like you need another action rather than a getter

**Reply by zh-pan** [reply]
Thanks reply.
In this case, I want cache some 'like button' object in order to use later, and the buttons should have some event when it clicked, so, the event maybe is a action will better.
by the way, if I use a action to get this buttons, every time will return diffrent objects, itsn't my wish.
